src/views/AddItem.jsx

```diff
@@ -1,26 +1,38 @@
 import React, { useEffect, useState } from 'react';
 import { addItem } from '../api/firebase';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, data }) {
 	const [itemName, setItemName] = useState('');
 	const [anticipation, setAnticipation] = useState('7');
-	const [message, setMessage] = useState('');
+	const [message, setMessage] = useState(null);
 
 	const handleSubmit = async (event) => {
 		event.preventDefault();
 		const daysUntilNextPurchase = parseInt(anticipation);
 
-		try {
-			await addItem(listToken, {
-				itemName,
-				daysUntilNextPurchase,
-			});
-
-			setMessage(`Item '${itemName}' was saved to the database.`);
-			setItemName('');
-			setAnticipation('7');
-		} catch (error) {
-			setMessage(`Error adding item`);
+		const itemNames = data.map((item) =>
+			item.name.toLowerCase().replace(/[^a-z0-9]/gi, ''),
+		);
+
+		if (itemName === '') {
+			setMessage('Please enter the name of your item.');
+		} else if (
+			itemNames.includes(itemName.toLowerCase().replace(/[^a-z0-9]/gi, ''))
```

Comment:
Nit - The way that you did it is totally awesome and works. 

Another way to do it that to me feels a little easier to read would be to use `.some` rather than `.map`. This is just because in my mind the code would read a little more like natural language.

Using early returns could also help you to avoid unnecessary iterations by not performing something like `.map` or `.some` if you know that it won't be needed b/c `itemName` is an empty string.

```javascript
if (itemName?.trim() === '') {
  setMessage('Please enter the name of your item.');
  // returning early will make it so that nothing else in the function executes
  return;
};

const isAlreadyInList = data.some(item => {
   return itemName.toLowerCase().trim() === item?.toLowerCase().trim()
})

if (isAlreadyInList) {
   setMessage(`${itemName} is already on the list.`);
  return;
}

try {
// API call and whatnot
```